### ISSUE TITLE
As a developer I want to have confidence that code dependencies are thoroughly tested and have clearly defined interfaces and predictable behavior.

### ISSUE THREAD

**Issue by maniarathi** [chanzuckerberg/single-cell#48]
**Story**: As a developer, I need to spend less time debugging to figure out why my change might have caused a deep-dependency change. I want confidence that each layer's interfaces have been thoroughly tested, especially around the edge-cases, so that if I create a new module that depends on it, I have confidence that I have not introduced a dependency bug that is hard to debug and takes up a lot of time. These errors also pass down to our desktop users who report back bugs that generally get added directly to our Sprint pipeline and cause expensive context shifts during a Sprint, reducing our velocity.
**Need**: Developers need thorough unit tests so that behavior of modules is explainable, predictable, and (more) bug-free. This will also inform us when we have introduced code that might have broken cellxgene in certain circumstances that would later be potentially discovered by the scientist using the desktop version.

**TL;DR**: Improve unit test coverage in cellxgene from 64% to 90%.
